### PR TITLE
FEAT:  API for update OOO requests

### PR DESCRIPTION
**Pull request by sahsisunny**
## Issue Ticket Number
- https://github.com/Real-Dev-Squad/website-backend/issues/1877
## Description

In this PR added a new model, middleware, and controller to update the OOO request.
### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [x] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

image

image

</details>

## Test Coverage

- https://github.com/Real-Dev-Squad/website-backend/pull/1879
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

File: types/oooRequest.d.ts [hunk 1]
```diff
@@ -26,6 +26,24 @@ export type OooStatusRequestBody = {
...
+
+export type OooRequestUpdateBody = {
+  lastModifiedBy?: string;
+  type?: string;
+  id?: string;
+  reason?: string;
+  state: REQUEST_STATE.APPROVED | REQUEST_STATE.REJECTED;
+  updatedAt?: admin.firestore.Timestamp;
+};
+
```
Comment (ajoykumardas12) [inline types/oooRequest.d.ts:46]: This is same type twice

File: types/oooRequest.d.ts [hunk 2]
```diff
@@ -26,6 +26,24 @@ export type OooStatusRequestBody = {
...
 };
 
+export type OooRequestUpdateBody = {
+  lastModifiedBy?: string;
+  type?: string;
+  id?: string;
+  reason?: string;
+  state: REQUEST_STATE.APPROVED | REQUEST_STATE.REJECTED;
+  updatedAt?: admin.firestore.Timestamp;
+};
```
Comment (ajoykumardas12) [inline types/oooRequest.d.ts:36]: Is everything optional except state? 
Also, type had specific strings as types in design docs I think.

File: models/oooRequests.ts [hunk 3]
```diff
@@ -38,4 +38,39 @@ export const createOooRequest = async (body: OooStatusRequestBody) => {
     logger.error(ERROR_WHILE_CREATING_OOO_REQUEST, error);
     throw error;
   }
-};
\ No newline at end of file
+};
+
+export const updateOooRequest = async (id: string, body: OooRequestUpdateBody, lastModifiedBy:string) => {
+  try {
+    const requestExists = await oooRequestModel.doc(id).get();
```
Comment (ajoykumardas12) [inline models/oooRequests.ts:45]: NIT: this variable name can be better, this is request object/document from firestore.

Comment (sahsisunny) [reply models/oooRequests.ts]: Done

File: types/oooRequest.d.ts (hunk 2 above)
Comment (sahsisunny) [reply types/oooRequest.d.ts:36]: its depend on the usage for middleware state is mandatory, for model updatedAt is optional.

Comment (ajoykumardas12) [reply types/oooRequest.d.ts:36]: Design doc mentions `type` property to have type as specific string only.

Screenshot_8

Comment (sahsisunny) [reply types/oooRequest.d.ts]: added

File: controllers/oooRequests.ts [hunk 4]
```diff
@@ -77,26 +77,7 @@ export const updateOooRequestController = async (req: OooRequestUpdateRequest, r
...
-            };
-            await addLog(oooRequestLog.type, oooRequestLog.meta, oooRequestLog.body);
-            return res.status(201).json({
-                message: OOO_STATUS_REQUEST_UPDATED_SUCCESSFULLY,
-                data: {
-                    id: oooRequestId,
-                    ...oooRequestResult,
-                },
-            });
-
```
Comment (ajoykumardas12) [inline controllers/oooRequests.ts]: Didn't get this change.

Comment (sahsisunny) [reply controllers/oooRequests.ts]: Not needed to log this

**Review by ajoykumardas12 — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**